StripeConnect/StripeConnect/Source/Internal/Webview/HTTPStatusError.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  HTTPStatusError.swift
+//  
+//
+//  Created by Mel Ludowise on 10/14/24.
+//
+
+import Foundation
+@_spi(STP) import StripeCore
+
+/// Error passed to the when the `ConnectComponentWebViewController.didFailLoadWithError`
+/// when receiving an error status code loading the component web page
+struct HTTPStatusError: Error, CustomNSError {
+    /// The HTTP status code
+    let errorCode: Int
+}
```

Comment:
Note: This is not public but it can potentially get passed to the integrator in `didFailLoadWithError`. I considered putting this through API review, but ultimately decided it was unnecessary since there's no value in the integrator explicitly checking for this error code. 

The error will have:
- Domain = `StripeConnect.HTTPStatusError`
- Code = the http status code
- Debug description = `Component loaded with HTTP status {errorCode}`
- Localized description = `There was an unexpected error -- try again in a few seconds`